stubs/whatthepatch/whatthepatch/patch.pyi

```diff
@@ -0,0 +1,77 @@
+from collections.abc import Iterator
+from typing import Any, NamedTuple
+
+from . import exceptions as exceptions
+from .snippets import findall_regex as findall_regex, split_by_regex as split_by_regex
+
+class header(NamedTuple):
+    index_path: str | None
+    old_path: str
+    old_version: str | None
+    new_path: str
+    new_version: str | None
+
+class diffobj(NamedTuple):
+    header: header | None
+    changes: list[Change] | None
+    text: str
+
+class Change(NamedTuple):
+    old: int | None
+    new: int | None
+    line: int | None
+    hunk: int
+
+file_timestamp_str: str
+diffcmd_header: Any
+unified_header_index: Any
+unified_header_old_line: Any
+unified_header_new_line: Any
+unified_hunk_start: Any
+unified_change: Any
```

Comment:
These could be annotated as `Pattern[str]`, where `Pattern` is imported from `typing`.